catalog/models.py

```diff
@@ -1,3 +1,124 @@
+""" catalog models for data storage
+
+The models are defined as classes for every real world object needed in
+the system. Each model class needs to inherid from django.db.models.Model.
+The methods for CRUD (create, read, update and delete) are inherited from
+the super class.
+"""
+
+
 from django.db import models
+import uuid
+from django.db.models.deletion import SET_NULL
+from django.urls import reverse     # Used to generate URLs by reversing the URL patterns
+
+class Genre(models.Model):
+    """Model representing a book genre."""
+
+    name = models.CharField(max_length=200, help_text='Enter a book genre (e.g. Science Fiction)')
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.name
+
+class Language(models.Model):
```

Comment:
it's a good practice to inherit a model from your custom abstract base model or normal base model depend upon use case rather than models.Model.

Example.
class BaseModel(models.Model):
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        abstract = True

Reply:
I am not sure if I understand this properly. Can you elaborate a bit about what you mean by abstract base model? From what I can understand, an abstract base class that inherits from models.Model should be created (which can have some generic features like storing the time of creation) and all current models should inherit from this class. Is that correct?

Reply:
yes, all current models should inherit from your abstract base class model. **Abstract Base Classes** are useful when you want to put some common information into a number of other models. You write your base class and put abstract = True in the MetaClass.